### PR TITLE
Fix pcre jit leak

### PR DESCRIPTION
**Pull request by bazsi**
This branch builds on PR #1345 by @juhaszviktor so it works on older PCRE releases as well. It removes ifdef trickery by introducing a compat/pcre.h file.

### CODE REVIEW

**Comment by bazsi** [comment]
I've added the missing file from the tarball, so it should be green now. fingers crossed :)

File: lib/compat/pcre.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#define COMPAT_PCRE_H_INCLUDED
+
+#include "compat/compat.h"
+#include <pcre.h>
+
+#ifndef PCRE_CONFIG_JIT
+#define pcre_free_study pcre_free
+#endif
+
+#endif /* COMPAT_PCRE_H_INCLUDED */
```
Comment (lbudai) [inline lib/compat/pcre.h]: Adding this to compat is cool :)

**Comment by lbudai** [comment]
and now I will ask kira-syslog-ng to test this pr (the first time for me :) )

**Comment by lbudai** [comment]
@kira-syslogng retest this please

**Comment by lbudai** [comment]
@pzoleex : how can I run those tests? :)

**Comment by pzoleex** [comment]
@kira-syslogng retest this please

**Comment by pzoleex** [comment]
@lbudai first add jenkins-test label and then ask to test

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**